changelogs/fragments/461-collection-dir.yml

```diff
@@ -0,0 +1,4 @@
+removed_features:
+  - "The ``--collection-dir`` option was removed from the ``antsibull-build`` subcommands ``build-collection``, ``build-multiple``, ``build-single``,
+     ``changelog``, ``collection``, ``multiple``, ``new-acd``, ``new-ansible``, ``prepare``, ``rebuild-single``, and ``single``, which should never
+     have had this option. The option was not used for these subcommands (https://github.com/ansible-community/antsibull/pull/461)."
```

Comment:
I don't think any of those subcommands had a `--collection-dir` option. The other parent parsers and subcommands that inherit from `build_parser` were instantiated before  `--collection-dir` was added to `build_parser`. All this really does is adds `--collection-dir` to the `build-collection` subcommand (i.e. the only subparser that inherits from `collection_parser`).

Reply:
Ah, interesting. I would expect it to work differently, but I was able to verify what you wrote...